[GTK] Remove WebKitInstallMissingMediaPluginsPermissionRequest

philn opened the pull request:
#### 209971efaec52e671a78a608074d0224c9ef1b1a
<pre>
[GTK] Remove WebKitInstallMissingMediaPluginsPermissionRequest
<a href="https://bugs.webkit.org/show_bug.cgi?id=239132">https://bugs.webkit.org/show_bug.cgi?id=239132</a>

Reviewed by Michael Catanzaro.

This API wasn&apos;t triggered anymore since the WebCore GStreamer platform layer only advertises support
for codecs that are guaranteed to work.

* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::HTMLMediaElement::requestInstallMissingPlugins): Deleted.
* Source/WebCore/html/HTMLMediaElement.h:
* Source/WebCore/page/ChromeClient.h:
(WebCore::ChromeClient::requestInstallMissingMediaPlugins): Deleted.
* Source/WebCore/platform/GStreamer.cmake:
* Source/WebCore/platform/graphics/MediaPlayer.h:
(WebCore::MediaPlayerClient::requestInstallMissingPlugins): Deleted.
* Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp:
(WebCore::MediaPlayerPrivateGStreamer::~MediaPlayerPrivateGStreamer):
(WebCore::MediaPlayerPrivateGStreamer::handleMessage):
* Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.h:
* Source/WebCore/platform/graphics/gstreamer/MediaPlayerRequestInstallMissingPluginsCallback.h: Removed.
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/SourcesWPE.txt:
* Source/WebKit/UIProcess/API/glib/WebKitInstallMissingMediaPluginsPermissionRequest.cpp:
(webkitInstallMissingMediaPluginsPermissionRequestAllow):
(webkitInstallMissingMediaPluginsPermissionRequestDeny):
(webkit_install_missing_media_plugins_permission_request_get_description):
(createGstInstallPluginsContext): Deleted.
(webkitInstallMissingMediaPluginsPermissionRequestCreate): Deleted.
* Source/WebKit/UIProcess/API/glib/WebKitInstallMissingMediaPluginsPermissionRequest.h.in:
* Source/WebKit/UIProcess/API/glib/WebKitInstallMissingMediaPluginsPermissionRequestPrivate.h: Removed.
* Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
(webkitWebViewRequestInstallMissingMediaPlugins): Deleted.
* Source/WebKit/UIProcess/API/glib/WebKitWebViewPrivate.h:
* Source/WebKit/UIProcess/API/gtk/PageClientImpl.cpp:
(WebKit::PageClientImpl::decidePolicyForInstallMissingMediaPluginsPermissionRequest): Deleted.
* Source/WebKit/UIProcess/API/gtk/PageClientImpl.h:
* Source/WebKit/UIProcess/API/wpe/PageClientImpl.cpp:
(WebKit::PageClientImpl::decidePolicyForInstallMissingMediaPluginsPermissionRequest): Deleted.
* Source/WebKit/UIProcess/API/wpe/PageClientImpl.h:
* Source/WebKit/UIProcess/Launcher/glib/XDGDBusProxy.cpp:
(WebKit::XDGDBusProxy::dbusSessionProxy):
* Source/WebKit/UIProcess/PageClient.h:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/UIProcess/gstreamer/InstallMissingMediaPluginsPermissionRequest.cpp: Removed.
* Source/WebKit/UIProcess/gstreamer/InstallMissingMediaPluginsPermissionRequest.h: Removed.
* Source/WebKit/UIProcess/gstreamer/WebPageProxyGStreamer.cpp: Removed.
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
(WebKit::WebChromeClient::requestInstallMissingMediaPlugins): Deleted.
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.h:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::close):
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:
* Source/WebKit/WebProcess/WebPage/gstreamer/WebPageGStreamer.cpp: Removed.
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebExtensions.cpp:
(beforeAll):
(permissionRequestCallback): Deleted.
(testInstallMissingPluginsPermissionRequest): Deleted.
* Tools/TestWebKitAPI/glib/TestExpectations.json:

Canonical link: <a href="https://commits.webkit.org/258653@main">https://commits.webkit.org/258653@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/62e11a2d0f63f8af09ca410be67219b1d2e35427

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/102130 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/11275 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/35202 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/111456 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/171631 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/106112 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/12254 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/2188 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/94511 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/109193 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/107909 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/9383 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/92659 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/37194 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/91252 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/24140 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/78920 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/4829 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/25564 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/4943 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/2002 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/10998 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/45060 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5939 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/6688 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro commented on Source/WebKit/UIProcess/API/glib/WebKitInstallMissingMediaPluginsPermissionRequest.cpp:
```diff
@@ -152,18 +111,14 @@ WebKitInstallMissingMediaPluginsPermissionRequest* webkitInstallMissingMediaPlug
-    request->priv->description = description.utf8();
-#endif
-    return request->priv->description.data();
+    return nullptr;
```
This isn't nullable, so you need to return an empty string "" instead.

mcatanzaro commented on Source/WebKit/UIProcess/API/glib/WebKitInstallMissingMediaPluginsPermissionRequest.cpp:
```diff
@@ -132,15 +100,6 @@ static void webkit_install_missing_media_plugins_permission_request_class_init(W
     objectClass->dispose = webkitInstallMissingMediaPluginsPermissionRequestDispose;
```
You can get rid of the dispose func now, since it just calls deny and that doesn't do anything anymore.

mcatanzaro commented on Source/WebKit/UIProcess/API/glib/WebKitInstallMissingMediaPluginsPermissionRequest.cpp:
```diff
@@ -152,18 +111,14 @@ WebKitInstallMissingMediaPluginsPermissionRequest* webkitInstallMissingMediaPlug
  * Returns: a string with the description provided by the media backend.
```
Another example of how I like to do doc comments for lobotomized APIs:

```
/**
 * webkit_install_missing_media_plugins_permission_request_get_description:
 * @request: a #WebKitInstallMissingMediaPluginsPermissionRequest
 *
 * This function returns an empty string.
 *
 * Returns: an empty string
 *
 * Since: 2.10
 *
 * Deprecated: 2.40
 */
 ```

mcatanzaro commented on Source/WebKit/UIProcess/API/glib/WebKitInstallMissingMediaPluginsPermissionRequest.cpp:
```diff
@@ -51,14 +41,13 @@ using namespace WebCore;
-    RefPtr<InstallMissingMediaPluginsPermissionRequest> request;
-#endif
     CString description;
     bool madeDecision;
```
I don't think these are needed anymore either, so you should be able to remove them.

mcatanzaro commented on Source/WebKit/UIProcess/API/glib/WebKitInstallMissingMediaPluginsPermissionRequest.cpp:
```diff
@@ -51,14 +41,13 @@ using namespace WebCore;
  * Since: 2.10
+ *
+ * Deprecated: 2.40
  */
```
Would be good to update this documentation to indicate that it no longer does anything, in addition to marking it as deprecated. Something like:

```
/**
 * WebKitInstallMissingMediaPluginsPermissionRequest:
 * @See_also: #WebKitPermissionRequest, #WebKitWebView
 *
 * Previously, a permission request for installing missing media plugins.
 *
 * WebKitInstallMissingMediaPluginsPermissionRequest will no longer ever be created, so
 * you can remove any code that attempts to handle it.
 *
 * Since: 2.10
 *
 * Deprecated: 2.40
 */
```

mcatanzaro commented on Source/WebKit/UIProcess/Launcher/glib/XDGDBusProxy.cpp:
```diff
@@ -69,9 +69,7 @@ std::optional<CString> XDGDBusProxy::dbusSessionProxy(const char* baseDirectory,
         CString(dbusAddress), m_dbusSessionProxyPath,
-        "--filter",
-        // GStreamers plugin install helper.
-        "--call=org.freedesktop.PackageKit=org.freedesktop.PackageKit.Modify2.InstallGStreamerResources@/org/freedesktop/PackageKit"
```
Nice attention to detail...

mcatanzaro reviewed (approved):
Excellent, thanks! One fewer thing to worry about, and you got to delete more than I expected.

philn committed:
[GTK] Remove WebKitInstallMissingMediaPluginsPermissionRequest
https://bugs.webkit.org/show_bug.cgi?id=239132

Reviewed by Michael Catanzaro.

This API wasn't triggered anymore since the WebCore GStreamer platform layer only advertises support
for codecs that are guaranteed to work.

* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::HTMLMediaElement::requestInstallMissingPlugins): Deleted.
* Source/WebCore/html/HTMLMediaElement.h:
* Source/WebCore/page/ChromeClient.h:
(WebCore::ChromeClient::requestInstallMissingMediaPlugins): Deleted.
* Source/WebCore/platform/GStreamer.cmake:
* Source/WebCore/platform/graphics/MediaPlayer.h:
(WebCore::MediaPlayerClient::requestInstallMissingPlugins): Deleted.
* Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp:
(WebCore::MediaPlayerPrivateGStreamer::~MediaPlayerPrivateGStreamer):
(WebCore::MediaPlayerPrivateGStreamer::handleMessage):
* Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.h:
* Source/WebCore/platform/graphics/gstreamer/MediaPlayerRequestInstallMissingPluginsCallback.h: Removed.
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/SourcesWPE.txt:
* Source/WebKit/UIProcess/API/glib/WebKitInstallMissingMediaPluginsPermissionRequest.cpp:
(webkitInstallMissingMediaPluginsPermissionRequestAllow):
(webkitInstallMissingMediaPluginsPermissionRequestDeny):
(webkit_install_missing_media_plugins_permission_request_get_description):
(createGstInstallPluginsContext): Deleted.
(webkitInstallMissingMediaPluginsPermissionRequestCreate): Deleted.
* Source/WebKit/UIProcess/API/glib/WebKitInstallMissingMediaPluginsPermissionRequest.h.in:
* Source/WebKit/UIProcess/API/glib/WebKitInstallMissingMediaPluginsPermissionRequestPrivate.h: Removed.
* Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
(webkitWebViewRequestInstallMissingMediaPlugins): Deleted.
* Source/WebKit/UIProcess/API/glib/WebKitWebViewPrivate.h:
* Source/WebKit/UIProcess/API/gtk/PageClientImpl.cpp:
(WebKit::PageClientImpl::decidePolicyForInstallMissingMediaPluginsPermissionRequest): Deleted.
* Source/WebKit/UIProcess/API/gtk/PageClientImpl.h:
* Source/WebKit/UIProcess/API/wpe/PageClientImpl.cpp:
(WebKit::PageClientImpl::decidePolicyForInstallMissingMediaPluginsPermissionRequest): Deleted.
* Source/WebKit/UIProcess/API/wpe/PageClientImpl.h:
* Source/WebKit/UIProcess/Launcher/glib/XDGDBusProxy.cpp:
(WebKit::XDGDBusProxy::dbusSessionProxy):
* Source/WebKit/UIProcess/PageClient.h:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/UIProcess/gstreamer/InstallMissingMediaPluginsPermissionRequest.cpp: Removed.
* Source/WebKit/UIProcess/gstreamer/InstallMissingMediaPluginsPermissionRequest.h: Removed.
* Source/WebKit/UIProcess/gstreamer/WebPageProxyGStreamer.cpp: Removed.
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
(WebKit::WebChromeClient::requestInstallMissingMediaPlugins): Deleted.
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.h:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::close):
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:
* Source/WebKit/WebProcess/WebPage/gstreamer/WebPageGStreamer.cpp: Removed.
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebExtensions.cpp:
(beforeAll):
(permissionRequestCallback): Deleted.
(testInstallMissingPluginsPermissionRequest): Deleted.
* Tools/TestWebKitAPI/glib/TestExpectations.json:

Canonical link: https://commits.webkit.org/258653@main
